src/mc/mcsim.f03

```diff
@@ -129,6 +136,96 @@ pure function list_confinement()
               endif
           endif
 
+        ! set wlc_basepairs to prop if not slide move
+        if (MCTYPE/=13) then 
+            wlc_basepairs_prop = wlc_basepairs
+        endif
+
+         ! sterics check here !
+          left = minval(wlc_pointsMoved(1:wlc_nPointsMoved))
+          right = maxval(wlc_pointsMoved(1:wlc_nPointsMoved))
+          if(WLC_P__GJK_STERICS) then
+            call MC_sterics(collisions,left,right,netSterics)
+            ! ascribe collision penalty
+            if (netSterics) then 
+                energyOf(sterics_)%dx = collisions 
+            else
+                if (collisions > 0) then 
+                    wlc_ATTEMPTS(MCTYPE) = wlc_ATTEMPTS(MCTYPE) + 1
+                    goto 10 ! skip move, return RP to nan
+                endif
+            endif
+          endif
+
+          ! internucleosome check here
+          if (WLC_P__INTERNUCLEOSOME_ON) then
+            delInt = 0
+            ! will only do internucleosome if sterics is on (probably)
+            do i = left,right
```

Comment:
Why use left and right rather than loop over wlc_pointsMoved?  The code used to have a left right structure but I moved to the pointsMoved methods because it allows for disjointed sections to move.  There is a bendpoints variable to keep track of the edge of the points moved.

Reply:
same reasoning as i said in the left/right for updating the positions of moved beads. let me know if you have a better suggestion--i'm all ears!